test/cpu_reduce_util.hpp

```diff
@@ -29,19 +29,117 @@
 #include <half.hpp>
 #include <limits>
 #include <cmath>
+#include <cassert>
+#include <stdexcept>
+#include <string>
 #include <miopen/miopen.h>
 #include <miopen/reduce_common.hpp>
 
-#include <miopen/bfloat16.hpp>
-
 namespace reduce {
 
+template <typename T>
+static inline bool float_equal_one(T);
+
+static inline bool float_equal_one(float x) { return x == 1.0f; };
+
+static inline bool float_equal_one(double x) { return x == 1.0; };
+
+static inline bool float_equal_one(half_float::half x)
+{
+    return x == convert_type<half_float::half>(1.0f);
+};
+
+template <typename T>
+static inline bool float_equal_zero(T x);
+
+static inline bool float_equal_zero(float x) { return x == 0.0f; };
+
+static inline bool float_equal_zero(double x) { return x == 0.0; };
+
+static inline bool float_equal_zero(half_float::half x)
+{
+    return x == convert_type<half_float::half>(0.0f);
+};
+
+template <typename T>
+static inline T Sqrt(T a);
+
+static inline float Sqrt(float a) { return sqrtf(a); };
+
+static inline double Sqrt(double a) { return sqrt(a); };
+
+static inline half_float::half Sqrt(half_float::half a) { return half_float::sqrt(a); };
+
+template <typename T>
+static inline T Abs(T a);
+
+static inline float Abs(float a) { return fabsf(a); };
+
+static inline double Abs(double a) { return fabs(a); };
+
+static inline half_float::half Abs(half_float::half a) { return half_float::abs(a); };
```

Comment:
`sqrt` and `abs` can be found through ADL as well, so you should do `using std::abs` and `using std::sqrt`.

Reply:
Yes.  I had ever though about that, but it takes me some time to confirm the availability of abs/sqrt for float/double in libstdc++-dev.     Corrected!  Thanks